rusoto/core/src/request.rs

```diff
@@ -75,21 +76,63 @@ impl From<IoError> for HttpDispatchError {
     }
 }
 
-///Trait for implementing HTTP Request/Response
+/// Trait for implementing HTTP Request/Response
 pub trait DispatchSignedRequest {
+    /// The future response value.
+    type Future: Future<Item=HttpResponse, Error=HttpDispatchError> + 'static;
     /// Dispatch Request, and then return a Response
-    fn dispatch(&self, request: &SignedRequest) -> Result<HttpResponse, HttpDispatchError>;
+    fn dispatch(&self, request: SignedRequest) -> Self::Future;
```

Comment:
Do we no longer take a reference to a `SignedRequest` because the underlying object is being moved to the reactor thread?

Reply:
Yep, that's basically right! Generally borrowing does not work well with futures, and we often want to move the arguments to a function into the future in order to use them there.